File: leetcode/0246-strobogrammatic-number/Solution.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public boolean isStrobogrammatic(String num) {
        Map<Character, Character> pairs = new HashMap<>();
        pairs.put('0', '0');
        pairs.put('1', '1');
        pairs.put('6', '9');
        pairs.put('8', '8');
        pairs.put('9', '6');

        int left = 0, right = num.length() - 1;
        while (left <= right) {
            char l = num.charAt(left);
            if (!pairs.containsKey(l) || pairs.get(l) != num.charAt(right)) {
                return false;
            }
            left++;
            right--;
        }

        return true;
    }
}

File: leetcode/0246-strobogrammatic-number/solution.cpp
#include <string>
#include <unordered_map>
using namespace std;

class Solution {
public:
    bool isStrobogrammatic(string num) {
        unordered_map<char, char> pairs = {
            {'0', '0'}, {'1', '1'}, {'6', '9'}, {'8', '8'}, {'9', '6'}
        };

        int left = 0, right = (int)num.size() - 1;
        while (left <= right) {
            auto it = pairs.find(num[left]);
            if (it == pairs.end() || it->second != num[right]) {
                return false;
            }
            left++;
            right--;
        }

        return true;
    }
};
